### PR TITLE
Add image transport plugins

### PR DESCRIPTION
**Pull request by PerrineAguiar**
This PR fixes some issues encountered when trying to use the image transport plugins (e.g. compressed image transport) on Android:
* It collects the missing image transport plugins
* It adds a patch to remove export of non existing plugin files
* It adds the share folder to the roscpp android ndk. The latter is needed because it contains some 3rd party libraries used by opencv in the compressed image transport plugin.

### CODE REVIEW

**Comment by PerrineAguiar** [comment]
@adamantivm @jubeira ptal

**Comment by ernestmc** [comment]
@PerrineAguiar thanks for your contribution. Could you add a very basic example like the ones for [pluginlib](https://github.com/ekumenlabs/roscpp_android/tree/master/files/pluginlib_sample_app) and [nodelets](https://github.com/ekumenlabs/roscpp_android/tree/master/files/nodelet_sample_app)? This will help for testing purposes and as a usage template.

File: do_everything.sh [hunk 1]
```diff
@@ -433,6 +435,23 @@ echo
 
 (cd $prefix/nodelet_sample_app && ant debug)
 
+echo
+echo -e '\e[34mCreating image transport sample app.\e[39m'
+echo
+# Copy opencv 3rdparty libraries. These are needed to build the compressed image transport plugin.
+( cp $prefix/roscpp_android_ndk/share/OpenCV/3rdparty/lib/* $prefix/roscpp_android_ndk/lib/)
```
Comment (PerrineAguiar) [inline do_everything.sh]: Is there a less dirty way to include these libraries without having to copy them to the lib folder?

**Comment by PerrineAguiar** [comment]
@ernestmc I added a sample app which tests the image transport plugin, please take a look.

File: do_everything.sh [hunk 2]
```diff
@@ -274,6 +274,8 @@ if [[ $skip -ne 1 ]] ; then
         apply_patch /opt/roscpp_android/patches/poco.patch
         # Patch pluginlib for static loading
         apply_patch /opt/roscpp_android/patches/pluginlib.patch
+	# Patch image_transport to fix faulty export plugins
+	apply_patch /opt/roscpp_android/patches/image_transport.patch
```
Comment (ernestmc) [inline do_everything.sh]: Please replace TAB with spaces to follow style.

File: do_everything.sh (hunk 1 above)
Comment (ernestmc) [inline do_everything.sh]: I think this way is good for now but I would put this code in the `build_library.sh` file with a conditional check for "opencv".

**Comment by PerrineAguiar** [comment]
@ernestmc Thanks for your review. I addressed your comments.

**Comment by PerrineAguiar** [comment]
@ernestmc Note that if you try to build the example again with the last changes, you will have to remove the existing libopencv_core.a first. Otherwise it won't copy the 3rdparty libs.

**Comment by ernestmc** [comment]
Thank you @PerrineAguiar ! I've tested again and it works perfect. Merging now.

**Comment by jubeira** [comment]
Great work @PerrineAguiar , thanks!

**Pull request merged**